docs: adds decision record about default datasource (#3346)
eclipse-edc/Connector: docs/developer/decision-records/2023-08-01-default-datasource/README.md

# Default Datasource configuration

## Decision

All SQL store extensions, if the specific datasource name is not configured should fall back to the default one.

## Rationale

> Each SQL store implementation as of today needs to have at least one mandatory configuration

Reviewer: ```suggestion
Each SQL store implementation currently must have at least one mandatory configuration
```

parameter `edc.datasource.<entity>.name` and then in case additional parameters as `url`, `user`, `password` and so on.

> This is powerful for fine-grained configuration and deployment. For simpler case and scenario all the store should

Reviewer: ```suggestion
This is powerful for fine-grained configuration and deployment. For simpler cases and scenario all the stores should
```

default to a common datasource name, that will be easier to configure.

## Approach

> All the SQL store extension will default to a common datasource name defined

Reviewer: ```suggestion
All the SQL store extensions will default to a common datasource name defined
```
